sstables/sstable_directory.cc

```diff
@@ -382,27 +382,44 @@ sstable_directory::retrieve_shared_sstables() {
     return std::exchange(_shared_sstable_info, {});
 }
 
+bool sstable_directory::compare_sstable_storage_prefix(const sstring& prefix_a, const sstring& prefix_b) noexcept {
+    size_t size_a = prefix_a.size();
+    if (prefix_a.back() == '/') {
```

Comment:
while ()? or use std::path::weakly_canonical()?

Reply:
The weakly_canonical doesn't trim trailing slashes.
Doing it in a loop ... tempting, but there are no places that produce it. And in fact, even _this_ is not bullet-proof, not even close to it, there can be (in theory) duplicating slashes in the middle of the path, but this check doesn't care

Reply:
It should be `std::filesystem::equivalent()` to be correct but that's too large hammer

Reply:
> The weakly_canonical doesn't trim trailing slashes.

https://godbolt.org/z/M59K3aWo3

Reply:
Alternatively, it can be 
```
    if (<this_check>()) {
        return true;
    } else {
        logger.warn("call support");
        return fs::equivalent();
    }
```
so that "this-check" thing covers 99.9999% of the cases